### PR TITLE
Use Jacobian Based Algorithms if user supplies custom Jacobian

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "NonlinearSolve"
 uuid = "8913a72c-1f9b-4ce2-8d82-65094dcecaec"
 authors = ["SciML"]
-version = "2.8.0"
+version = "2.9.0"
```
Comment (ChrisRackauckas) [inline Project.toml:4]: Did this need a minor version?

Comment (ChrisRackauckas) [inline Project.toml:4]:
```suggestion
version = "2.8.1"
```

Comment (avik-pal) [reply Project.toml]: No really patch is fine

**Comment by avik-pal** [comment]
I want to test this with BoundaryValueDiffEq once before merging

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**